Source/WebCore/html/parser/HTMLEntityParser.cpp

```diff
@@ -291,12 +291,11 @@ DecodedHTMLEntity consumeHTMLEntity(StringParsingBuffer<UChar>& source)
     return consumeHTMLEntity(StringParsingBufferSource<UChar> { source }, 0);
 }
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 DecodedHTMLEntity decodeNamedHTMLEntityForXMLParser(const char* name)
 {
     HTMLEntitySearch search;
-    while (*name) {
-        search.advance(*name++);
+    for (char character : span(name)) {
```

Comment:
I wonder if we should mark the `span(char*)` helper unsafe. According to -Wunsafe-buffer-usage-in-libc-call we should, because the type `char*` does not ensure null termination. 

(Doesn't block this patch, but maybe we should consider it.)